diff --git a/coregrind/m_syswrap/syswrap-x86-linux.c b/coregrind/m_syswrap/syswrap-x86-linux.c
index bec145a..11bb13d 100644
--- a/coregrind/m_syswrap/syswrap-x86-linux.c
+++ b/coregrind/m_syswrap/syswrap-x86-linux.c
@@ -1604,7 +1604,10 @@ static SyscallTableEntry syscall_table[] = {
    LINX_(__NR_sendmsg,           sys_sendmsg),          // 370
    LINXY(__NR_recvfrom,          sys_recvfrom),         // 371
    LINXY(__NR_recvmsg,           sys_recvmsg),          // 372
-   LINX_(__NR_shutdown,          sys_shutdown)          // 373
+   LINX_(__NR_shutdown,          sys_shutdown),         // 373
+
+   /* Explicitly not supported on i386 yet. */
+   GENX_(__NR_arch_prctl,        sys_ni_syscall)        // 384
 };
 
 SyscallTableEntry* ML_(get_linux_syscall_entry) ( UInt sysno )
commit 12d162aabc8180d9ad0c095e72a1354e93b091d7
Author: Tom Hughes <tom@compton.nu>
Date:   Wed Mar 21 12:28:31 2018 +0000

    Update linux system call numbers.

diff --git a/include/vki/vki-scnums-x86-linux.h b/include/vki/vki-scnums-x86-linux.h
index 8a6376b..041d247 100644
--- a/include/vki/vki-scnums-x86-linux.h
+++ b/include/vki/vki-scnums-x86-linux.h
@@ -32,7 +32,7 @@
 #ifndef __VKI_SCNUMS_X86_LINUX_H
 #define __VKI_SCNUMS_X86_LINUX_H
 
-// Derived from linux-2.6/arch/x86/syscalls/syscall_32.tbl
+// Derived from linux-2.6/arch/x86/entry/syscalls/syscall_32.tbl
 
 #define __NR_restart_syscall      0
 #define __NR_exit		  1
@@ -411,6 +411,14 @@
 #define __NR_userfaultfd        374
 #define __NR_membarrier         375
 #define __NR_mlock2             376
+#define __NR_copy_file_range    377
+#define __NR_preadv2            378
+#define __NR_pwritev2           379
+#define __NR_pkey_mprotect      380
+#define __NR_pkey_alloc         381
+#define __NR_pkey_free          382
+#define __NR_statx              383
+#define __NR_arch_prctl         384
 
 #endif /* __VKI_SCNUMS_X86_LINUX_H */
 
