diff --git a/Makefile.all.am b/Makefile.all.am
index 3786e34..1befef5 100644
--- a/Makefile.all.am
+++ b/Makefile.all.am
@@ -50,20 +50,20 @@ inplace-noinst_DSYMS: build-noinst_DSYMS
 	done
 
 # This is used by coregrind/Makefile.am and by <tool>/Makefile.am for doing
-# "make install".  It copies $(noinst_PROGRAMS) into $prefix/lib/valgrind/.
+# "make install".  It copies $(noinst_PROGRAMS) into $prefix/libexec/valgrind/.
 # It needs to be depended on by an 'install-exec-local' rule.
 install-noinst_PROGRAMS: $(noinst_PROGRAMS)
-	$(mkinstalldirs) $(DESTDIR)$(pkglibdir); \
+	$(mkinstalldirs) $(DESTDIR)$(pkglibexecdir); \
 	for f in $(noinst_PROGRAMS); do \
-	  $(INSTALL_PROGRAM) $$f $(DESTDIR)$(pkglibdir); \
+	  $(INSTALL_PROGRAM) $$f $(DESTDIR)$(pkglibexecdir); \
 	done
 
 # This is used by coregrind/Makefile.am and by <tool>/Makefile.am for doing
-# "make uninstall".  It removes $(noinst_PROGRAMS) from $prefix/lib/valgrind/.
+# "make uninstall".  It removes $(noinst_PROGRAMS) from $prefix/libexec/valgrind/.
 # It needs to be depended on by an 'uninstall-local' rule.
 uninstall-noinst_PROGRAMS:
 	for f in $(noinst_PROGRAMS); do \
-	  rm -f $(DESTDIR)$(pkglibdir)/$$f; \
+	  rm -f $(DESTDIR)$(pkglibexecdir)/$$f; \
 	done
 
 # Similar to install-noinst_PROGRAMS.
@@ -71,15 +71,15 @@ uninstall-noinst_PROGRAMS:
 # directories.  XXX: not sure whether the resulting permissions will be
 # correct when using 'cp -R'...
 install-noinst_DSYMS: build-noinst_DSYMS
-	$(mkinstalldirs) $(DESTDIR)$(pkglibdir); \
+	$(mkinstalldirs) $(DESTDIR)$(pkglibexecdir); \
 	for f in $(noinst_DSYMS); do \
-	  cp -R $$f.dSYM $(DESTDIR)$(pkglibdir); \
+	  cp -R $$f.dSYM $(DESTDIR)$(pkglibexecdir); \
 	done
 
 # Similar to uninstall-noinst_PROGRAMS.
 uninstall-noinst_DSYMS:
 	for f in $(noinst_DSYMS); do \
-	  rm -f $(DESTDIR)$(pkglibdir)/$$f.dSYM; \
+	  rm -f $(DESTDIR)$(pkglibexecdir)/$$f.dSYM; \
 	done
 
 # This needs to be depended on by a 'clean-local' rule.
diff --git a/Makefile.am b/Makefile.am
index 242b38a..3b7c806 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -58,7 +58,7 @@ DEFAULT_SUPP_FILES = @DEFAULT_SUPP@
 # default.supp, as it is built from the base .supp files at compile-time.
 dist_noinst_DATA = $(SUPP_FILES)
 
-vglibdir = $(pkglibdir)
+vglibdir = $(pkglibexecdir)
 vglib_DATA = default.supp
 
 pkgconfigdir = $(libdir)/pkgconfig
diff --git a/coregrind/Makefile.am b/coregrind/Makefile.am
index 94030fd..f09763a 100644
--- a/coregrind/Makefile.am
+++ b/coregrind/Makefile.am
@@ -11,12 +11,12 @@ include $(top_srcdir)/Makefile.all.am
 
 AM_CPPFLAGS_@VGCONF_PLATFORM_PRI_CAPS@ += \
 	-I$(top_srcdir)/coregrind \
-	-DVG_LIBDIR="\"$(pkglibdir)"\" \
+	-DVG_LIBDIR="\"$(pkglibexecdir)"\" \
 	-DVG_PLATFORM="\"@VGCONF_ARCH_PRI@-@VGCONF_OS@\""
 if VGCONF_HAVE_PLATFORM_SEC
 AM_CPPFLAGS_@VGCONF_PLATFORM_SEC_CAPS@ += \
 	-I$(top_srcdir)/coregrind \
-	-DVG_LIBDIR="\"$(pkglibdir)"\" \
+	-DVG_LIBDIR="\"$(pkglibexecdir)"\" \
 	-DVG_PLATFORM="\"@VGCONF_ARCH_SEC@-@VGCONF_OS@\""
 endif
 
@@ -714,7 +714,7 @@ GDBSERVER_XML_FILES = \
 	m_gdbserver/mips64-fpu.xml
 
 # so as to make sure these get copied into the install tree
-vglibdir = $(pkglibdir)
+vglibdir = $(pkglibexecdir)
 vglib_DATA  = $(GDBSERVER_XML_FILES)
 
 # so as to make sure these get copied into the tarball
diff --git a/mpi/Makefile.am b/mpi/Makefile.am
index 7ad9a25..471fee0 100644
--- a/mpi/Makefile.am
+++ b/mpi/Makefile.am
@@ -18,16 +18,18 @@ EXTRA_DIST = \
 # libmpiwrap-<platform>.so
 #----------------------------------------------------------------------------
 
-noinst_PROGRAMS  =
+# These are really real libraries, so they should go to libdir, not libexec.
+mpidir = $(pkglibdir)
+mpi_PROGRAMS  =
 if BUILD_MPIWRAP_PRI
-noinst_PROGRAMS += libmpiwrap-@VGCONF_ARCH_PRI@-@VGCONF_OS@.so
+mpi_PROGRAMS += libmpiwrap-@VGCONF_ARCH_PRI@-@VGCONF_OS@.so
 endif
 if BUILD_MPIWRAP_SEC
-noinst_PROGRAMS += libmpiwrap-@VGCONF_ARCH_SEC@-@VGCONF_OS@.so
+mpi_PROGRAMS += libmpiwrap-@VGCONF_ARCH_SEC@-@VGCONF_OS@.so
 endif
 
 if VGCONF_OS_IS_DARWIN
-noinst_DSYMS = $(noinst_PROGRAMS)
+mpi_DSYMS = $(mpi_PROGRAMS)
 endif
 
 
